kamon-core/src/main/scala/kamon/trace/Span.scala

```diff
@@ -16,155 +16,178 @@
-    if(sampled && isOpen)
-      logs = Span.LogEntry(timestampMicroseconds, fields) :: logs
-    this
-  }
+trait BaseSpan {
 
-  override def log(timestampMicroseconds: Long, fields: java.util.Map[String, _]): Span =
-    log(timestampMicroseconds, fields.asScala.asInstanceOf[Map[String, _]])
+  def context(): SpanContext
 
-  override def log(event: String): Span = synchronized {
-    if(sampled && isOpen)
-      logs = Span.LogEntry(Clock.microTimestamp(), Map("event" -> event)) :: logs
-    this
-  }
+  def capture(): Continuation
 
-  override def log(timestampMicroseconds: Long, event: String): Span = synchronized {
-    if(sampled && isOpen)
-      logs = Span.LogEntry(timestampMicroseconds, Map("event" -> event)) :: logs
-    this
-  }
+  def annotate(annotation: Span.Annotation): Span
 
-  override def log(eventName: String, payload: scala.Any): Span = synchronized {
-    if(sampled && isOpen)
-      logs = Span.LogEntry(Clock.microTimestamp(), Map(eventName -> payload)) :: logs
-    this
-  }
+  def addSpanTag(key: String, value: String): Span
+
+  def addMetricTag(key: String, value: String): Span
+
+  def addBaggage(key: String, value: String): Span
+
+  def getBaggage(key: String): Option[String]
+
+  def setOperationName(name: String): Span
 
-  override def log(timestampMicroseconds: Long, eventName: String, payload: scala.Any): Span = synchronized {
-    if(sampled && isOpen)
-      logs = Span.LogEntry(timestampMicroseconds, Map(eventName -> payload)) :: logs
-    this
+  def disableMetricsCollection(): Span
+
+  def finish(): Unit
+
+  def finish(finishTimestampMicros: Long): Unit
+
+}
+
+object Span {
+
+  final class Empty(tracer: Tracer) extends Span {
+    override val context: SpanContext = SpanContext.EmptySpanContext
+    override def capture(): Continuation = Continuation.Default(this, tracer)
+
+    override def annotate(annotation: Annotation): Span = this
+    override def addSpanTag(key: String, value: String): Span = this
+    override def addMetricTag(key: String, value: String): Span = this
+    override def addBaggage(key: String, value: String): Span = this
+    override def getBaggage(key: String): Option[String] = None
+    override def setOperationName(name: String): Span = this
+    override def disableMetricsCollection(): Span = this
+    override def finish(): Unit = {}
+    override def finish(finishTimestampMicros: Long): Unit = {}
   }
 
-  override def getBaggageItem(key: String): String =
-    spanContext.getBaggage(key)
+  object Empty {
+    def apply(tracer: Tracer): Empty = new Empty(tracer)
+  }
 
-  override def context(): SpanContext =
-    spanContext
+  /**
+    *
+    * @param spanContext
+    * @param initialOperationName
+    * @param initialTags
+    * @param startTimestampMicros
+    * @param reporterRegistry
+    */
+  final class Real(spanContext: SpanContext, initialOperationName: String, initialTags: Map[String, String],
```

Comment:
why would that be slightly better? Once again, the idea I have is that the JVM can optimize synchronized on this better and there is no contention expected for Spans, we just need safety :D